cmspage/models.py

```diff
@@ -24,68 +35,235 @@ class Meta:
         proxy = True
 
 
-class AbstractCMSPage(Page):
+class SiteVariables(models.Model):
```

Comment:
Maybe this format of caching is a reasonable approach, but the presented solution has an issue because it is a static method, it does not take the current site (set via wagtail Site as determined by the request) into consideration. Site-specific variables are different per site, which is the entire point of this class.
The cache needs to contain a different key for each site.

Reply:
yes, this is certainly an improvement, please provide a diff